example/sparse/matrix_factorization/data.py

```diff
@@ -18,38 +18,9 @@
 import os, gzip
 import sys
 import mxnet as mx
-
-class DummyIter(mx.io.DataIter):
-    "A dummy iterator that always return the same batch, used for speed testing"
-    def __init__(self, real_iter):
-        super(DummyIter, self).__init__()
-        self.real_iter = real_iter
-        self.provide_data = real_iter.provide_data
-        self.provide_label = real_iter.provide_label
-        self.batch_size = real_iter.batch_size
-
-        for batch in real_iter:
-            self.the_batch = batch
-            break
-
-    def __iter__(self):
-        return self
-
-    def next(self):
-        return self.the_batch
-
-def get_libsvm_data(data_dir, data_name, url):
-    if not os.path.isdir(data_dir):
-        os.mkdir(data_dir)
-    os.chdir(data_dir)
-    if (not os.path.exists(data_name)):
-        print("Dataset " + data_name + " not present. Downloading now ...")
-        import urllib
-        zippath = os.path.join(data_dir, data_name + ".bz2")
-        urllib.urlretrieve(url + data_name + ".bz2", zippath)
-        os.system("bzip2 -d %r" % data_name + ".bz2")
-        print("Dataset " + data_name + " is now present.")
-    os.chdir("..")
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.append(os.path.join(curr_path, "../../../tests/python/common"))
```

Comment:
Nitpick: This creates a dependency on the other test modules in repo. It would be nice if end users could just download the examples and run it on a mxnet installed system.

Reply:
I don't know what's a good solution. Functions such as `GetMNIST_ubyte` is used in many examples, do we really want to have separate copies of this function in each example? But your concern is valid, I've seen users posting issues regarding the path problem. 
@szha

Reply:
If it is a function that is going to be used by many examples shouldn't it go with the python modules in mxnet python package. For eg. inside test_utils?

Reply:
Maybe move them to test_utils? @eric-haibin-lin

Reply:
sounds good. will do